hikari/commands.py

```diff
@@ -124,8 +124,10 @@ class CommandOption:
         This will be inclusively between 1-100 characters in length.
     """
 
-    is_required: bool = attr.field(repr=False)
-    """Whether this command """
+    is_required: typing.Optional[bool] = attr.field(repr=False)
+    """Whether this command is required. This will be `builtins.None` if the option type
+    is `SUB_COMMAND` or `SUB_COMMAND_GROUP`.
+    """
```

Comment:
```suggestion
    is_required: typing.Optional[bool] = attr.field(repr=False)
    """Whether this command is required.
    
    This will be `builtins.None` if the option type is `SUB_COMMAND`
    or `SUB_COMMAND_GROUP`.
    """
```

Reply:
That isn't how the API object works, not present means false for all command option types as per the docs and it'll  never be returned as null. this should just be defaulting to False as it is in #924